react/week3/todo-list/src/App.js

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+import TodoList from "./TodoList";
+import Fetching from "./Fetching";
+import { getNextId } from "./Helper";
+import Timer from "./Timer";
+import PropTypes from "prop-types";
+
+function App() {
+  const [todoText, setTodoText] = useState("");
+  const [deadline, setDeadline] = useState("");
+  const [todos, setTodos] = useState([]);
+
+  useEffect(() => {
+    Fetching().then((data) => {
+      setTodos(data);
+    });
+  }, []);
+
+  const addTodo = () => {
+    if (!todoText) {
+      return alert("Insert new todo");
+    }
+    if (!deadline) {
+      return alert("insert deadline");
+    }
```

Comment:
Nice that you have handled the empty cases. I would probably make the feedback to the user a little more informative about how they made an error. Something like "Todo title cannot be empty - please insert a todo title" or whatever you think is appropriate.